src/sinks/new_relic/service.rs

```diff
@@ -0,0 +1,124 @@
+use crate::{
+    http::HttpClient,
+    sinks::util::Compression
+};
+use hyper::Body;
+use tracing::Instrument;
+use vector_core::{
+    stream::DriverResponse,
+    event::{
+        EventStatus, Finalizable, EventFinalizers
+    },
+    internal_event::EventsSent,
+    buffers::Ackable
+};
+use futures::future::BoxFuture;
+use http::{
+    Request,
+    header::{
+        CONTENT_ENCODING, CONTENT_LENGTH, CONTENT_TYPE
+    }
+};
+use std::{
+    fmt::Debug,
+    task::{
+        Context, Poll
+    },
+    sync::Arc
+};
+use tower::Service;
+use super::{
+    NewRelicCredentials, NewRelicSinkError
+};
+
+#[derive(Debug, Clone)]
+pub struct NewRelicApiRequest {
+    pub batch_size: usize,
+    pub finalizers: EventFinalizers,
+    pub credentials: Arc<NewRelicCredentials>,
+    pub payload: Vec<u8>,
+    pub compression: Compression
+}
+
+impl Ackable for NewRelicApiRequest {
+    fn ack_size(&self) -> usize {
+        self.batch_size
+    }
+}
+
+impl Finalizable for NewRelicApiRequest {
+    fn take_finalizers(&mut self) -> EventFinalizers {
+        std::mem::take(&mut self.finalizers)
+    }
+}
+
+#[derive(Debug)]
+pub struct NewRelicApiResponse {
+    event_status: EventStatus,
+    count: usize,
+    events_byte_size: usize
+}
+
+impl DriverResponse for NewRelicApiResponse {
+    fn event_status(&self) -> EventStatus {
+        self.event_status
+    }
+
+    fn events_sent(&self) -> EventsSent {
+        EventsSent {
+            count: self.count,
+            byte_size: self.events_byte_size,
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct NewRelicApiService {
+    pub client: HttpClient
+}
+
+impl Service<NewRelicApiRequest> for NewRelicApiService {
+    type Response = NewRelicApiResponse;
+    type Error = NewRelicSinkError;
+    type Future = BoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    fn poll_ready(&mut self, _cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
+
+    fn call(&mut self, request: NewRelicApiRequest) -> Self::Future {
+
+        info!("Sending {} events.", request.batch_size);
```

Comment:
This should probably be a lower log level, `debug` or `trace`

Reply:
Resolved in https://github.com/vectordotdev/vector/pull/9764/commits/4ece2dd9fae8de86089e76dc2bce4db4ba51bf2a